Rockets booking and cancelation

VicPeralta opened the pull request:
# Rockets booking and cancelation

In this milestone:

- When the user clicks the "Reserve rocket" button, an action is dispatched to update the store. The state is not mutated.
- All logic is in the reducer.
- When the user clicks the "Cancel reservation" button, an action is dispatched to update the store. The state is not mutated.
- The list of reserved rockets is stored in the local storage, in order to persist the data of reserved rockets.

leopiresgoss reviewed (changes requested):
**Hello, @VicPeralta**  :slightly_smiling_face: 

I couldn't check if it is working because the 'Fetching data' is not being removed.
I tried to find the issue, but I couldn't find it as it doesn't show any error.

![image](https://user-images.githubusercontent.com/61767582/161834532-850f53ae-87a9-47d4-918c-69f96e0c4358.png)

Happy coding :tada:

VicPeralta commented:
> **Hello, @VicPeralta** 🙂
> 
> I couldn't check if it is working because the 'Fetching data' is not being removed. I tried to find the issue, the problem might be with the split method according to the second image
> 
> ![image](https://user-images.githubusercontent.com/61767582/161834532-850f53ae-87a9-47d4-918c-69f96e0c4358.png)
> 
> ![image](https://user-images.githubusercontent.com/61767582/161836160-01e06977-0fb6-4cb9-b9e2-b4d3e21bbaa7.png)
> 
> Happy coding 🎉

Hello @leopiresgoss 
Thank you for pointing out this error, it happened because I didn't check a possible undefined value when the local storage is empty, as I implemented first the option to save, and therefore the data was already there. I fixed it. Could you please check it again?

leopiresgoss reviewed (approved):
Good job @VicPeralta :1st_place_medal:

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

Congratulations and Happy Coding :tada:
